torch/csrc/jit/passes/prepack_folding.cpp

```diff
@@ -0,0 +1,59 @@
+#include <stack>
+
+#include <torch/csrc/jit/api/module.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/prepack_folding.h>
+
+namespace torch {
+namespace jit {
+
+// Must run this pass after constant folding.
+void FoldPrePackingOps(script::Module& m,
+    const PrePackingOpsFilterFn& is_foldable_op) {
+  // Since this pass can be called by quantization or other passes as well,
+  // we need to make sure we generate a unique "packed_weight_\d+"
+  // Thus static uid.
+  static int64_t uid = 0;
+
+  auto method = m.get_method("forward");
```

Comment:
Why just forward?  Can we do all methods?

Reply:
As mentioned in the previous comment, this will be called after inlining so only forward method is what should matter.

Reply:
What about models with multiple entry points?

Reply:
If you mean by, executing model by `model.get_method("foo")(inputs)` then yes, this will fail. The reason for not extending this beyond `forward` was mainly because, this pass relies on freezing pass, which freezes only forward method of the module to be frozen. But this can be fixed.